goss-testing/tests/common/goss-interface-has-ip.yaml

```diff
@@ -29,10 +29,16 @@ command:
     meta:
       desc: Validates that interface '{{$interface}}' has an IP address
       sev: 0
-    exec: "ip addr show dev {{$interface}} scope global | grep 'inet ' | awk '{print $2}'"
+    exec: |-
+      if ! eval ip addr show dev {{$interface}} 1>/dev/null; then
+        if [[ "{{$interface}}" == "bond0.can0" ]]; then
+          echo "{{$interface}} is optional...skipping"
+        else
+          ip addr show dev {{$interface}} scope global | grep 'inet ' | awk '{print $2}'
+        fi
+      fi
     exit-status: 0
     stdout:
-    - /[1-9]/
```

Comment:
If you're removing this line, how are you detecting failures? The above script is always going to exit with a 0 return code, unless I'm missing something.

Reply:
If you just want to skip the test for that interface, you could use conditional execution in goss. Something like this:
```
command:
{{range $interface := index .Vars "network_interfaces"}}
    {{if ne {{$interface}} "bond0.can0"}}
        check_interface_ip_{{$interface}}:
            title: Interface '{{$interface}}' Has IP
            meta:
                desc: Validates that interface '{{$interface}}' has an IP address
                sev: 0
            exec: |-
                set -eo pipefail
                ip addr show dev {{$interface}} scope global | grep inet' | awk '{print $2}'
            exit-status: 0
            stdout:
                - /[1-9]/
            timeout: 10000
            skip: false
    {{end}}
{{end}}
```

Reply:
Oh slick. Must have overlooked that capability in the docs.  Makes sense though

Reply:
I think I mistakenly deleted that stdout..I'll fix this up like the other one

Reply:
Actually, I was just thinking this will always skip the can.  If it is there, we do want to check it... we might need to re-think this a bit

Reply:
It's easy to overlook stuff in the goss docs, I think. I don't find them particularly well organized. :) 

In that case, I think you will need to have the logic in the test script itself. I don't know a way in goss to run a command on the system and, based on its output or return code, conditionally execute a test or not. So I think you will need to do something like your solution.

Reply:
OK I updated the logic here to fail gracefully if can0 is not there:
```
    {{if eq $interface "bond0.can0"}}
    # it's ok if the bond0.can0 fails since it is optional
    exec: "if ! eval ip -4 a s dev {{$interface}} | awk '/inet / {print $2}'; then : ; fi"
    {{else}}
    exec: "ip -4 a s dev {{$interface}} | awk '/inet / {print $2}'"
    {{end}}
```